src/main/java/org/radarbase/ksql/udaf/MedianUdaf.java

```diff
@@ -0,0 +1,38 @@
+package org.radarbase.ksql.udaf;
+
+import io.confluent.ksql.function.udaf.Udaf;
+import io.confluent.ksql.function.udaf.UdafDescription;
+import io.confluent.ksql.function.udaf.UdafFactory;
+import java.util.List;
+import org.apache.commons.math3.stat.StatUtils;
+import org.apache.kafka.connect.data.Struct;
+
+@UdafDescription(name = "median",
+        author = "yatharthranjan",
+        version = "1.0.0",
+        description = "Calculates the median of numbers in a stream.")
+public class MedianUdaf {
+
+    private MedianUdaf() {
+
+    }
+
+    @UdafFactory(
+            description = "Calculates the median of values in a stream.",
+            aggregateSchema = "STRUCT<SAMPLES ARRAY<double>, COUNT bigint>"
+    )
+    public static Udaf<Double, Struct, Double> createUdaf() {
+        return new MedianUdafImpl();
+    }
+
+    private static class MedianUdafImpl extends UniformSamplingReservoirDoubleUdaf {
+
+        @Override
+        public Double map(Struct agg) {
+            List<Double> samples = agg.getArray(UniformSamplingReservoirDoubleUdaf.SAMPLES);
+            if (samples.isEmpty()) return 0.0;
```

Comment:
If the sample is empty, we should return NaN since median does not exist in that case.

Reply:
Good spot, I forgot doing that. I initially did indeed plan to return null and that's why chose the `Double` class instead of the `double` primitive.